vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -65,38 +56,45 @@ class OidcSiopWallet(
      * Implementations need to fetch the url passed in, and return either the body, if there is one,
      * or the HTTP header `Location`, i.e. if the server sends the request object as a redirect.
      */
-    private val remoteResourceRetriever: RemoteResourceRetrieverFunction = { null },
+    private val remoteResourceRetriever: RemoteResourceRetrieverFunction,
     /**
      * Need to verify the request object serialized as a JWS,
      * which may be signed with a pre-registered key (see [OpenIdConstants.ClientIdScheme.PRE_REGISTERED]).
      */
-    private val requestObjectJwsVerifier: RequestObjectJwsVerifier = RequestObjectJwsVerifier { _, _ -> true },
+    private val requestObjectJwsVerifier: RequestObjectJwsVerifier,
     /**
      * Need to implement if the presentation definition needs to be derived from a scope value.
      * See [ScopePresentationDefinitionRetriever] for implementation instructions.
      */
-    private val scopePresentationDefinitionRetriever: ScopePresentationDefinitionRetriever? = null,
+    private val scopePresentationDefinitionRetriever: ScopePresentationDefinitionRetriever,
 ) {
     companion object {
         fun newDefaultInstance(
-            keyPairAdapter: KeyPairAdapter = RandomKeyPairAdapter(),
-            holder: Holder = HolderAgent(keyPairAdapter),
-            jwsService: JwsService = DefaultJwsService(DefaultCryptoService(keyPairAdapter)),
-            clock: Clock = Clock.System,
-            clientId: String = "https://wallet.a-sit.at/",
-            remoteResourceRetriever: RemoteResourceRetrieverFunction = { null },
-            requestObjectJwsVerifier: RequestObjectJwsVerifier = RequestObjectJwsVerifier { jws, authnRequest -> true },
-            scopePresentationDefinitionRetriever: ScopePresentationDefinitionRetriever? = { null },
-        ) = OidcSiopWallet(
-            holder = holder,
-            agentPublicKey = keyPairAdapter.publicKey,
-            jwsService = jwsService,
-            clock = clock,
-            clientId = clientId,
-            remoteResourceRetriever = remoteResourceRetriever,
-            requestObjectJwsVerifier = requestObjectJwsVerifier,
-            scopePresentationDefinitionRetriever = scopePresentationDefinitionRetriever,
-        )
+            keyPairAdapter: KeyPairAdapter? = null,
```

Comment:
I am not a fan of making this nullable and then instantiating a random key pair by default. it was clearer before, imho

Reply:
See https://github.com/a-sit-plus/kmm-vc-library/pull/67#discussion_r1610210556
I can still change it back of course if the provided argument isn't satisfactory

Reply:
fixed in commit e0d9d1f448615405bd526b2e676580e2b90ae884